### PR TITLE
Report divulgence warning at commit location.

### PR DESCRIPTION
**Pull request by sofiafaro-da**
This PR records the commitLocation when the warning is generated, and uses it when creating the diagnostic.

Part of #9947, follow up from #10289

### CODE REVIEW

File: compiler/damlc/daml-ide-core/src/Development/IDE/Core/Rules/Daml.hs [hunk 1]
```diff
@@ -922,6 +925,26 @@ toDiagnostic file world range = \case
...
 
+    fileRangeFromMaybeLocation :: Maybe SS.Location -> (NormalizedFilePath, Range)
+    fileRangeFromMaybeLocation mbLocation =
+        fromMaybe (scenarioFile, scenarioRange) $ do
+            location <- mbLocation
+            lfModule <- LF.lookupLocationModule world location
+            filePath <- LF.moduleSource lfModule
+            Just (toNormalizedFilePath' filePath, rangeFromLocation location)
+
+    rangeFromLocation :: SS.Location -> LSP.Range
```
Comment (cocreature) [inline compiler/damlc/daml-ide-core/src/Development/IDE/Core/Rules/Daml.hs:936]: completely orthogonal but we should also be able to use the commit location for errors during scenario interpretation that happen in a `submit`. I’ll open an issue.

**Review by cocreature — APPROVED** [review]
Nice, thank you! Can we change the tests you added before to match on the location?

File: compiler/damlc/tests/daml-test-files/Divulgence.daml [hunk 2]
```diff
@@ -44,7 +44,8 @@ template DisclosureDelegation
         do
           fetch secretCid -- actor will divulge secretCid to p
 
--- @WARN range=48:1-48:5; Use of divulged contracts is deprecated
+-- @WARN range=49:1-49:5; Use of divulged contracts is deprecated
+-- @WARN range=49:1-49:5; Use of divulged contracts is deprecated
```
Comment (sofiafaro-da) [inline compiler/damlc/tests/daml-test-files/Divulgence.daml:48]: This isn't the submit location :-(

File: compiler/damlc/daml-ide-core/src/Development/IDE/Core/Rules/Daml.hs (hunk 1 above)
Comment (sofiafaro-da) [reply compiler/damlc/daml-ide-core/src/Development/IDE/Core/Rules/Daml.hs:936]: That's reasonable.

Also, I don't know exactly why, but this isn't working right now. I don't know if the commit location is off, or the source location, or something else.

Comment (cocreature) [reply compiler/damlc/daml-ide-core/src/Development/IDE/Core/Rules/Daml.hs:936]: Let me know if you want me to take a closer look or need a rubber duck for debugging

Comment (sofiafaro-da) [reply compiler/damlc/daml-ide-core/src/Development/IDE/Core/Rules/Daml.hs:936]: It looks like the commit location wasn't being set during a submit. I changed it, and now the locations show up correctly. (I think I still have an off by one error in the column, but that's easy to fix.)

Comment (cocreature) [reply compiler/damlc/daml-ide-core/src/Development/IDE/Core/Rules/Daml.hs:936]: uh that’s not good, nice job catching this

File: daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Speedy.scala [hunk 3]
```diff
@@ -798,7 +803,7 @@ private[lf] object Speedy {
             .initial(pkg2TxVersion, contractKeyUniqueness, submissionTime, initialSeeding),
           committers = committers,
           readAs = readAs,
-          commitLocation = None,
+          commitLocation = commitLocation,
```
Comment (cocreature) [inline daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Speedy.scala:806]: Nice :+1:

Comment (cocreature) [reply daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Speedy.scala:806]: I checked why this didn’t blow up before and it turns out we didn’t actually use commitLocation for anything. It was originally used before I split scenario interpretation into off-ledger and on-ledger interpretation but with the split we had access to it in `ScenarioRunner` already.

So at least nothing else was broken by the fact that we didn’t set this.

Comment (sofiafaro-da) [reply daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Speedy.scala:806]: Ok, good ... I was a bit worried.

**Pull request merged**